### ISSUE TITLE
NSUrlSessionHandler.RemoveNotification () - SIGABRT: Value cannot be null. Parameter name: observer

### ISSUE THREAD

**Issue by ideveloperaz** [xamarin/xamarin-macios#7152]
yes another crash in foundation:

NSUrlSessionHandler.RemoveNotification () - SIGABRT: Value cannot be null. Parameter name: observer

**Comment by ideveloperaz** [comment]
![Screen Shot 2019-10-02 at 8 31 29 AM](https://user-images.githubusercontent.com/2921595/66018441-24ab4980-e4f0-11e9-9ddc-17ffd329a3ad.png)

**Comment by ideveloperaz** [comment]
also during app running in debugger I get the following error from time to time:

![Screen Shot 2019-10-02 at 8 30 38 AM](https://user-images.githubusercontent.com/2921595/66018469-3e4c9100-e4f0-11e9-9a58-4bb029151a16.png)

**Comment by ideveloperaz** [comment]
these all are critical issue. we could not use normally simulator and app always crashes due to NSUrlSessionHandler.RemoveNotification () from time to time

**Comment by chamons** [comment]
Thank you for your feedback!

For us to investigate this further, could you please provide your full build logs, crash reports (if any), test case (to reproduce) and all your version information.

To get full build logs just set the log verbosity to diagnostic at the following locations:

- On Visual Studio for Mac: Preferences > Projects > Build
- On Visual Studio for Windows: Tools > Options > Projects and Solutions > Build and Run

On Visual Studio for Windows you also want to add `-v -v -v -v` to the mtouch additional arguments by right-clicking the project in the solution explorer and selecting `Properties`.

Note: this is done automatically on Visual Studio for Mac when the log verbosity is set to diagnostic.

The easiest way to get exact version information:

- On Visual Studio for Mac: "Visual Studio" menu, "About Visual Studio" item, "Show Details" button.
- On Visual Studio for Windows: "Help menu", "About Microsoft Visual Studio" item.

Then copy/paste the version information (you can use the "Copy Information" button).

We look forward to hearing from you!

**Comment by ideveloperaz** [comment]
Hello dear @chamons  ! thank you for quick response, here is needed details:

=== Visual Studio Community 2019 for Mac ===

Version 8.3.1 (build 18)
Installation UUID: f3992657-2718-41e7-955c-06f496e0aa5a
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.16.1.24 (d16-3 / 08809f5b)

	Package version: 604000198

=== Mono Framework MDK ===

Runtime:
	Mono 6.4.0.198 (2019-06/fe64a4765e6) (64-bit)
	Package version: 604000198

=== NuGet ===

Version: 5.3.0.6192

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/3.0.100/Sdks
SDK Versions:
	3.0.100
	2.1.701
	2.1.700
	2.1.505
	2.1.504
	2.1.503
	2.1.302
	2.1.301
	2.1.4
	2.0.0
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.0.0
	2.1.13
	2.1.12
	2.1.11
	2.1.9
	2.1.8
	2.1.7
	2.1.2
	2.1.1
	2.0.5
	2.0.0

=== Xamarin.Profiler ===

Version: 1.6.12.26
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 11.0 (14936)
Build 11A420a

=== Xamarin.Mac ===

Version: 6.2.0.42 (Visual Studio Community)
Hash: 5e8a208b
Branch: d16-3
Build date: 2019-09-17 14:19:30-0400

=== Xamarin.iOS ===

Version: 13.2.0.42 (Visual Studio Community)
Hash: 5e8a208b
Branch: d16-3
Build date: 2019-09-17 14:19:30-0400

=== Xamarin Designer ===

Version: 16.3.0.244
Hash: e15b2ee97
Branch: remotes/origin/d16-3
Build date: 2019-09-25 00:03:16 UTC

=== Xamarin.Android ===

Version: 10.0.0.43 (Visual Studio Community)
Commit: xamarin-android/d16-3/8af1ca8
Android SDK: /Users/rufat/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		6.0 (API level 23)
		7.1 (API level 25)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.2
SDK Build Tools Version: 28.0.3

Build Information: 
Mono: mono/mono/2019-06@7af64d1ebe9
Java.Interop: xamarin/java.interop/d16-3@5836f58
LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
ProGuard: xamarin/proguard/master@905836d
SQLite: xamarin/sqlite/3.27.1@8212a2d
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/rufat/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 1.4.0.65
Hash: c33b107
Branch: remotes/origin/d16-3
Build date: 2019-09-19 20:42:44 UTC

=== Android Device Manager ===

Version: 1.2.0.115
Hash: 724ea69
Branch: remotes/origin/d16-3
Build date: 2019-09-19 20:43:06 UTC

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 803010018
Git revision: 272ab73f00ab2b13ec97dc8b31d02980cf66d47f
Build date: 2019-09-27 19:26:51+00
Build branch: release-8.3
Xamarin extensions: 451b00e65052c11c16b79e6d31a24c4cf0d1b794

=== Operating System ===

Mac OS X 10.14.6
Darwin 18.7.0 Darwin Kernel Version 18.7.0
    Tue Aug 20 16:57:14 PDT 2019
    root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64

**Comment by ideveloperaz** [comment]
I could not reproduce as it happens sporadically ... NSUrlSessionHandler.RemoveNotification crash comes from production app and customers (observed in appcenter) and "NSNotificationCenter.g.cs file not found" during app run in ios simulator during debug mode

**Comment by ideveloperaz** [comment]
Here is one of crashes:

**Comment by ideveloperaz** [comment]
Incident Identifier: 7d74390f-34f9-4437-9ef6-f2523a4bc80f
CrashReporter Key:   E109C819-4118-4B89-A47D-F2A644E0249F
Hardware Model:      iPhone7,2
Process:         testcompany.App.iOS [1347]
Path:            /var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/testcompany.App.iOS
Identifier:      no.testcompany.testcompany-app.dev
Version:         6.3.3 (1.5.12)
Code Type:       arm64
Parent Process:  ??? [1]

Date/Time:       undefined
Launch Time:     undefined
OS Version:      iPhone OS 12.4.1 (16G102)
Report Version:  104

Exception Type:  SIGABRT
Exception Codes: #0 at 0x1b447f0dc
Crashed Thread:  26

Application Specific Information:
*** Terminating app due to uncaught exception 'SIGABRT', reason: 'Value cannot be null.
Parameter name: observer'

Xamarin Exception Stack:
System.ArgumentNullException: Value cannot be null.
Parameter name: observer
  at System.Net.Http.NSUrlSessionHandler.RemoveNotification () <0x1017b90a8 + 0x0008c> in <4aba469338f14d9287166bf556c9e128#df13ae0a93942e859a1fdb19dd70d96b>:0
  at System.Net.Http.NSUrlSessionHandler.RemoveInflightData (Foundation.NSUrlSessionTask task, System.Boolean cancel) <0x1017b9358 + 0x0010f> in <4aba469338f14d9287166bf556c9e128#df13ae0a93942e859a1fdb19dd70d96b>:0
  at System.Net.Http.NSUrlSessionHandler+NSUrlSessionHandlerDelegate.DidCompleteWithError (Foundation.NSUrlSession session, Foundation.NSUrlSessionTask task, Foundation.NSError error) <0x1017b9d34 + 0x000c3> in <4aba469338f14d9287166bf556c9e128#df13ae0a93942e859a1fdb19dd70d96b>:0

Thread 26 Crashed:
0   libsystem_kernel.dylib               0x00000001b447f0dc __pthread_kill + 8
1   libsystem_c.dylib                    0x00000001b43d7ea8 abort + 136
2   testcompany.App.iOS                       0x00000001038dcf84 0x100e84000 + 44404612
3   testcompany.App.iOS                       0x00000001038d5644 0x100e84000 + 44373572
4   Mono                                 0x000000010531e324 0x10527c000 + 664356
5   Mono                                 0x00000001052b2f4c 0x10527c000 + 225100
6   Mono                                 0x00000001052b39a8 0x10527c000 + 227752
7   Mono                                 0x00000001052a6ad0 0x10527c000 + 174800
8   testcompany.App.iOS                       0x00000001014d8828 0x100e84000 + 6637608
9   testcompany.App.iOS                       0x00000001017b955c 0x100e84000 + 9655644
10  Mono                                 0x00000001052c520c 0x10527c000 + 299532
11  Mono                                 0x0000000105382c20 0x10527c000 + 1076256
12  Mono                                 0x00000001053864bc 0x10527c000 + 1090748
13  testcompany.App.iOS                       0x0000000100f2a9fc 0x100e84000 + 682492
14  testcompany.App.iOS                       0x0000000100f2a6fc 0x100e84000 + 681724
15  CFNetwork                            0x00000001b5066420 __51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.182 + 248
16  Foundation                           0x00000001b5347cf4 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 12
17  Foundation                           0x00000001b5253dc0 -[NSBlockOperation main] + 68
18  Foundation                           0x00000001b52532a8 -[__NSOperationInternal _start:] + 736
19  Foundation                           0x00000001b5349a78 __NSOQSchedule_f + 268
20  libdispatch.dylib                    0x00000001b4320a38 _dispatch_call_block_and_release + 20
21  libdispatch.dylib                    0x00000001b43217d4 _dispatch_client_callout + 12
22  libdispatch.dylib                    0x00000001b42c601c _dispatch_continuation_pop$VARIANT$mp + 408
23  libdispatch.dylib                    0x00000001b42c56e0 _dispatch_async_redirect_invoke + 596
24  libdispatch.dylib                    0x00000001b42d2030 _dispatch_root_queue_drain + 368
25  libdispatch.dylib                    0x00000001b42d28d4 _dispatch_worker_thread2 + 124
26  libsystem_pthread.dylib              0x00000001b45021b4 _pthread_wqthread + 460
27  libsystem_pthread.dylib              0x00000001b4504cd4 start_wqthread + 0

Thread 0:
0   libsystem_kernel.dylib               0x00000001b44740f4 mach_msg_trap + 8
1   CoreFoundation                       0x00000001b4874120 __CFRunLoopServiceMachPort + 232
2   CoreFoundation                       0x00000001b486f030 __CFRunLoopRun + 1356
3   CoreFoundation                       0x00000001b486e7c0 CFRunLoopRunSpecific + 432
4   GraphicsServices                     0x00000001b6a6f79c GSEventRunModal + 100
5   UIKitCore                            0x00000001e11f3c38 UIApplicationMain + 208
6   testcompany.App.iOS                       0x000000010179fbb0 0x100e84000 + 9550768
7   testcompany.App.iOS                       0x00000001017d6afc 0x100e84000 + 9775868
8   Mono                                 0x00000001052c520c 0x10527c000 + 299532
9   Mono                                 0x0000000105382c20 0x10527c000 + 1076256
10  Mono                                 0x0000000105388ee8 0x10527c000 + 1101544
11  Mono                                 0x00000001052a28d8 0x10527c000 + 157912
12  testcompany.App.iOS                       0x00000001038db9d4 0x100e84000 + 44399060
13  testcompany.App.iOS                       0x0000000100f3e908 0x100e84000 + 764168
14  libdyld.dylib                        0x00000001b43328e0 start + 0

Thread 1:
0   libsystem_kernel.dylib               0x00000001b447eee4 __psynch_cvwait + 8
1   Mono                                 0x0000000105429b0c 0x10527c000 + 1760012
2   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
3   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
4   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 2:
0   libsystem_kernel.dylib               0x00000001b4474130 semaphore_wait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 3:
0   libsystem_kernel.dylib               0x00000001b44740f4 mach_msg_trap + 8
1   CoreFoundation                       0x00000001b4874120 __CFRunLoopServiceMachPort + 232
2   CoreFoundation                       0x00000001b486f030 __CFRunLoopRun + 1356
3   CoreFoundation                       0x00000001b486e7c0 CFRunLoopRunSpecific + 432
4   Foundation                           0x00000001b523c98c -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 296
5   Foundation                           0x00000001b523c81c -[NSRunLoop(NSRunLoop) runUntilDate:] + 92
6   UIKitCore                            0x00000001e12d9754 -[UIEventFetcher threadMain] + 132
7   Foundation                           0x00000001b53694a0 __NSThread__start__ + 980
8   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
9   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
10  libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 4:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 5:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 6:
0   libsystem_kernel.dylib               0x00000001b44740f4 mach_msg_trap + 8
1   CoreFoundation                       0x00000001b4874120 __CFRunLoopServiceMachPort + 232
2   CoreFoundation                       0x00000001b486f030 __CFRunLoopRun + 1356
3   CoreFoundation                       0x00000001b486e7c0 CFRunLoopRunSpecific + 432
4   Foundation                           0x00000001b523c98c -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 296
5   libAXSpeechManager.dylib             0x00000001c458994c -[AXSpeechThread main] + 260
6   Foundation                           0x00000001b53694a0 __NSThread__start__ + 980
7   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
8   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
9   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 7:
0   libsystem_kernel.dylib               0x00000001b44740f4 mach_msg_trap + 8
1   CoreFoundation                       0x00000001b4874120 __CFRunLoopServiceMachPort + 232
2   CoreFoundation                       0x00000001b486f030 __CFRunLoopRun + 1356
3   CoreFoundation                       0x00000001b486e7c0 CFRunLoopRunSpecific + 432
4   AVFAudio                             0x00000001ba7e0334 GenericRunLoopThread::Entry(void*) + 152
5   AVFAudio                             0x00000001ba80ac60 CAPThread::Entry(CAPThread*) + 84
6   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
7   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
8   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 8:
0   libsystem_kernel.dylib               0x00000001b44740f4 mach_msg_trap + 8
1   CoreFoundation                       0x00000001b4874120 __CFRunLoopServiceMachPort + 232
2   CoreFoundation                       0x00000001b486f030 __CFRunLoopRun + 1356
3   CoreFoundation                       0x00000001b486e7c0 CFRunLoopRunSpecific + 432
4   Foundation                           0x00000001b523c98c -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 296
5   libAXSpeechManager.dylib             0x00000001c458994c -[AXSpeechThread main] + 260
6   Foundation                           0x00000001b53694a0 __NSThread__start__ + 980
7   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
8   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
9   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 9:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 10:
0   libsystem_kernel.dylib               0x00000001b447eee4 __psynch_cvwait + 8
1   Mono                                 0x000000010543a2b8 0x10527c000 + 1827512
2   Mono                                 0x00000001053dcff0 0x10527c000 + 1445872
3   Mono                                 0x00000001053dcec8 0x10527c000 + 1445576
4   Mono                                 0x00000001053dd100 0x10527c000 + 1446144
5   Mono                                 0x00000001053cc024 0x10527c000 + 1376292
6   Mono                                 0x0000000105344974 0x10527c000 + 821620
7   testcompany.App.iOS                       0x00000001014145ac 0x100e84000 + 5834156
8   testcompany.App.iOS                       0x00000001015d6278 0x100e84000 + 7676536
9   testcompany.App.iOS                       0x00000001015d5e2c 0x100e84000 + 7675436
10  testcompany.App.iOS                       0x00000001015d1618 0x100e84000 + 7656984
11  testcompany.App.iOS                       0x00000001015cfdb0 0x100e84000 + 7650736
12  Mono                                 0x00000001052c520c 0x10527c000 + 299532
13  Mono                                 0x0000000105382c20 0x10527c000 + 1076256
14  Mono                                 0x00000001053885c4 0x10527c000 + 1099204
15  Mono                                 0x00000001053d19a8 0x10527c000 + 1399208
16  Mono                                 0x00000001053d176c 0x10527c000 + 1398636
17  libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
18  libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
19  libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 11:
0   libsystem_kernel.dylib               0x00000001b447f328 __select + 8
1   Mono                                 0x00000001053c2bb4 0x10527c000 + 1338292
2   Mono                                 0x00000001053c24ac 0x10527c000 + 1336492
3   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
4   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
5   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
6   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
7   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 12:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 13:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 14:
0   libsystem_kernel.dylib               0x00000001b44740f4 mach_msg_trap + 8
1   CoreFoundation                       0x00000001b4874120 __CFRunLoopServiceMachPort + 232
2   CoreFoundation                       0x00000001b486f030 __CFRunLoopRun + 1356
3   CoreFoundation                       0x00000001b486e7c0 CFRunLoopRunSpecific + 432
4   CFNetwork                            0x00000001b4e8874c -[__CoreSchedulingSetRunnable runForever] + 212
5   Foundation                           0x00000001b53694a0 __NSThread__start__ + 980
6   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
7   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
8   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 15:
0   libsystem_kernel.dylib               0x00000001b447eee4 __psynch_cvwait + 8
1   testcompany.App.iOS                       0x00000001028dfd08 0x100e84000 + 27639048
2   testcompany.App.iOS                       0x00000001028dfb94 0x100e84000 + 27638676
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 16:
0   libsystem_kernel.dylib               0x00000001b44740f4 mach_msg_trap + 8
1   CoreFoundation                       0x00000001b4874120 __CFRunLoopServiceMachPort + 232
2   CoreFoundation                       0x00000001b486f030 __CFRunLoopRun + 1356
3   CoreFoundation                       0x00000001b486e7c0 CFRunLoopRunSpecific + 432
4   Foundation                           0x00000001b523c98c -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 296
5   libAXSpeechManager.dylib             0x00000001c458994c -[AXSpeechThread main] + 260
6   Foundation                           0x00000001b53694a0 __NSThread__start__ + 980
7   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
8   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
9   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 17:
0   libsystem_kernel.dylib               0x00000001b44740f4 mach_msg_trap + 8
1   CoreFoundation                       0x00000001b4874120 __CFRunLoopServiceMachPort + 232
2   CoreFoundation                       0x00000001b486f030 __CFRunLoopRun + 1356
3   CoreFoundation                       0x00000001b486e7c0 CFRunLoopRunSpecific + 432
4   TTSSpeechBundle                      0x000000010a39a9e4 0x10a378000 + 141796
5   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
6   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
7   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 18:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 19:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 20:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 21:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 22:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 23:
0   libsystem_pthread.dylib              0x00000001b4504cd0 start_wqthread + 0

Thread 24:
0   libsystem_kernel.dylib               0x00000001b447fb74 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001b4504cd4 start_wqthread + 0

Thread 25:
0   libsystem_kernel.dylib               0x00000001b447eee4 __psynch_cvwait + 8
1   Mono                                 0x000000010543a2b8 0x10527c000 + 1827512
2   Mono                                 0x00000001054428a0 0x10527c000 + 1861792
3   Mono                                 0x00000001053c5ac8 0x10527c000 + 1350344
4   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
5   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
6   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
7   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
8   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 27:
0   libsystem_pthread.dylib              0x00000001b4504cd0 start_wqthread + 0

Thread 28:
0   libsystem_kernel.dylib               0x00000001b447fb74 __workq_kernreturn + 8
1   libsystem_pthread.dylib              0x00000001b4504cd4 start_wqthread + 0

Thread 29:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 30:
0   libsystem_kernel.dylib               0x00000001b4474148 semaphore_timedwait_trap + 8
1   Mono                                 0x00000001053d18e8 0x10527c000 + 1399016
2   Mono                                 0x00000001053d176c 0x10527c000 + 1398636
3   libsystem_pthread.dylib              0x00000001b45012c0 _pthread_body + 124
4   libsystem_pthread.dylib              0x00000001b4501220 _pthread_start + 40
5   libsystem_pthread.dylib              0x00000001b4504cdc thread_start + 0

Thread 31:
0   libsystem_pthread.dylib              0x00000001b4504cd0 start_wqthread + 0

Thread 26 crashed with arm64 Thread State:
   x21: 0x0000000000000003     x2: 0x0000000000000000    x16: 0x0000000000000148     x3: 0x0000000115600000
   x22: 0x0000000000022527     x4: 0x000000011564ec20    x17: 0x0000000000000000   cpsr: 0x0000000000000000
    x5: 0x0000000000000032    x23: 0x000000016f64b0e0     x6: 0x0000000000000039    x18: 0x0000000000000000
   x10: 0x00000001b44f7f18     lr: 0x00000001b44f8094     x7: 0x0000000fffffffec    x24: 0x0000000105e51678
   x11: 0x00000001f050870b     x8: 0x0000000000000c00    x19: 0x0000000000000006    x25: 0x000000010834a5b8
    x9: 0x00000001b44fc888    x12: 0x00000001f050870b     fp: 0x000000016f649020    x26: 0x0000000105deee00
   x13: 0x0000000000000016     pc: 0x00000001b447f0dc    x27: 0x000000016f649e30    x14: 0x0000000000000001
   x20: 0x000000016f64b000     x0: 0x0000000000000000     sp: 0x000000016f648ff0    x28: 0x00000001014b12b3
   x15: 0x0000000000000881     x1: 0x0000000000000000

Binary Images:
0x0000000100e84000 -        0x0000000103fbbfff +testcompany.App.iOS arm64  <f5ddbe45eb2f3b15871bbe91947a0b52> /var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/testcompany.App.iOS
0x0000000104dc4000 -        0x0000000104dcbfff +mscodec2 arm64  <15803f3b96ba3d65a5b2b8348a18397a> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/mscodec2.framework/mscodec2
0x0000000104dd4000 -        0x0000000104e4bfff +bctoolbox arm64  <a16d137da0173fee9bf0aaaed668df3e> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/bctoolbox.framework/bctoolbox
0x0000000104e70000 -        0x0000000104e77fff +msopenh264 arm64  <339fca90052830cfa713a463fa082082> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/msopenh264.framework/msopenh264
0x0000000104e84000 -        0x0000000104e93fff +mediastreamer_base arm64  <9d57645cd7603eb98ac91deab05b8003> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/mediastreamer_base.framework/mediastreamer_base
0x0000000104eb0000 -        0x000000010506bfff +mediastreamer_voip arm64  <ae00653677383bd7aace4cb06aa3253b> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/mediastreamer_voip.framework/mediastreamer_voip
0x00000001050d0000 -        0x00000001050d7fff +msx264 arm64  <b531f8f102df31f6b8dc0d704749b8e7> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/msx264.framework/msx264
0x00000001050e4000 -        0x00000001050ebfff +msamr arm64  <f6283e2368283b62b40d304b49a67747> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/msamr.framework/msamr
0x00000001051c4000 -        0x00000001051effff +mssilk arm64  <26bbc2890ad63aef90921d62b0cdd53f> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/mssilk.framework/mssilk
0x0000000105200000 -        0x0000000105237fff +mswebrtc arm64  <0b3de9ec597d3e60ab5a67a018eb5961> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/mswebrtc.framework/mswebrtc
0x000000010524c000 -        0x0000000105267fff +ortp arm64  <72f404b26a2336d68b0288ebaac0f331> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/ortp.framework/ortp
0x000000010527c000 -        0x00000001054ebfff +Mono arm64  <77eab2849b80305b8506dc9c6a545195> /private/var/containers/Bundle/Application/1E3E19B2-AD8A-461F-99AC-2763086BF564/testcompany.App.iOS.app/Frameworks/Mono.framework/Mono
0x000000010a378000 -        0x000000010a3bbfff  TTSSpeechBundle arm64  <5ec074e6df15357ab5bc96c2f27c91b3> /System/Library/TTSPlugins/TTSSpeechBundle.speechbundle/TTSSpeechBundle
0x00000001b42c1000 -        0x00000001b4330fff  libdispatch.dylib arm64  <c3d9e9a5b56e3215aed59b6affc80d91> /usr/lib/system/libdispatch.dylib
0x00000001b4331000 -        0x00000001b435afff  libdyld.dylib arm64  <5e181e9004763f7ba24dc51d73effd19> /usr/lib/system/libdyld.dylib
0x00000001b437d000 -        0x00000001b43fefff  libsystem_c.dylib arm64  <fc4c83bb09133b92919b277e2b0d0ba7> /usr/lib/system/libsystem_c.dylib
0x00000001b445c000 -        0x00000001b4486fff  libsystem_kernel.dylib arm64  <451947374fc437afa258be0e4593163a> /usr/lib/system/libsystem_kernel.dylib
0x00000001b44f6000 -        0x00000001b4506fff  libsystem_pthread.dylib arm64  <7dc8076100ab3c07ba487f82aae5dbe4> /usr/lib/system/libsystem_pthread.dylib
0x00000001b47ca000 -        0x00000001b4b27fff  CoreFoundation arm64  <9a052c0830b6363babf2062f0b867758> /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation
0x00000001b4e87000 -        0x00000001b5234fff  CFNetwork arm64  <d160964071a233529369006ee750fa0d> /System/Library/Frameworks/CFNetwork.framework/CFNetwork
0x00000001b5235000 -        0x00000001b551ffff  Foundation arm64  <7a676a573fa63763a203969875bfaac3> /System/Library/Frameworks/Foundation.framework/Foundation
0x00000001b6a65000 -        0x00000001b6a77fff  GraphicsServices arm64  <5dc71f1e2db03637b0517c9574dd609e> /System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices
0x00000001ba754000 -        0x00000001ba84afff  AVFAudio arm64  <943ab4d1d6363303ac591636a41af0e7> /System/Library/Frameworks/AVFoundation.framework/Frameworks/AVFAudio.framework/AVFAudio
0x00000001c4582000 -        0x00000001c458cfff  libAXSpeechManager.dylib arm64  <7be784a209c432dd8f80036c37c190ad> /usr/lib/libAXSpeechManager.dylib
0x00000001e0937000 -        0x00000001e1a1bfff  UIKitCore arm64  <a1f463bfcf9b3796ba9e7c2b40617fae> /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore

**Comment by ideveloperaz** [comment]
here is debug logs:

[deb_logs.txt](https://github.com/xamarin/xamarin-macios/files/3682608/deb_logs.txt)

**Comment by ideveloperaz** [comment]
Hello @mandel-macaque  and @chamons  any updates here?

**Comment by uwe-neuronation** [comment]
Isn't that a duplicate of #6691 
But #6691 didn't make it to the stable release so fare, right?

**Comment by mandel-macaque** [comment]
@uwe-neuronation @ideveloperaz @chamons The fix was created in https://github.com/xamarin/xamarin-macios/commit/b062b4c5b6551be19f27149da31f0652540c3205 which was not back-ported to d16-3 or the Xcode 11 branch, that is the reason it does not work on 13.2.0.42.

Looking at git:
`git branch --contains b062b4c5b6551be19f27149da31f0652540c3205 -r`

Just returns that the commit is present in master and d16-4. The fix will be in d16-4 preview 2, which should be available this week. 

@chamons please take a look and decide if a back port is needed/worth it.

**Comment by mandel-macaque** [comment]
Closing issue, feel free to reopen, @chamons will decide if the back port is required taking into account the planned releases we have.

**Comment by chamons** [comment]
This will land in our Xcode 11.1 branch, and as @mandel-macaque said d16-4. I don't believe we'll have a service opportunity before Xcode 11.1 lands.

If people are interested, I can post a link to our Xcode 11.1 build once this lands, or you can use the preview later this week as well.

**Comment by mandel-macaque** [comment]
FYI back port can be found here: https://github.com/xamarin/xamarin-macios/pull/7184

**Comment by uwe-neuronation** [comment]
Is the fix included in the 13.4.0.2 release? It's based on d16-3, right? So it's not?

**Comment by rolfbjarne** [comment]
@uwe-neuronation the fix **is** included in the 13.4.0.2 release.
